### PR TITLE
[Feat] Add the option to interpolate the transport plan with the identity

### PR DESCRIPTION
**Pull request by pbarbarant**
This short PR implements an option to interpolate between the transport plan and the identity (as suggested by @bthirion ) when calling the ``.transform`` method. Such interpolation has shown significant improvements in decoding performances in my recent benchmarks, I thus propose to include it directly in FUGW.

### CODE REVIEW

File: src/fugw/mappings/dense.py [hunk 1]
```diff
@@ -270,7 +270,9 @@ def fit(
 
         return self
 
-    def transform(self, source_features, device="auto"):
+    def transform(
+        self, source_features, id_interpolation=False, device="auto"
```
Comment (bthirion) [inline src/fugw/mappings/dense.py:274]: I would call it regularization or shrinkage rather than interpolation

Comment (bthirion) [reply src/fugw/mappings/dense.py:274]: Also, I would use an explicit parameter rather than a Boolean. It could default to 0 (no regularization) or 0.5.

**Review by bthirion — COMMENTED** [review]
LGTM. 
I'd love to have a small examples showcasing the difference.

File: src/fugw/mappings/dense.py (hunk 1 above)
Comment (pbarbarant) [reply src/fugw/mappings/dense.py]: I changed the name of the parameter to ``id_reg`` for better clarity and it now defaults to zero.

**Comment by pbarbarant** [comment]
> LGTM. I'd love to have a small examples showcasing the difference.

I'm going to add some new plots at the end of the surfacic examples.

**Review by bthirion — COMMENTED** [review]
Looks good so far. The last missing points is a couple of tests to ensure that this behaves well.

**Comment by pbarbarant** [comment]
I added some tests, fixed a scalar factor, everything should be good to go.

File: src/fugw/mappings/sparse.py [hunk 2]
```diff
@@ -314,6 +314,10 @@ def transform(self, source_features, device="auto"):
         ----------
         source_features: ndarray(n_samples, n) or ndarray(n)
             Contrast map for source subject
+        id_reg: float, optional, defaults to 0
```
Comment (bthirion) [inline src/fugw/mappings/sparse.py:317]:
```suggestion
        id_reg: float, in the [0, 1] interval, defaults to 0
```

**Review by bthirion — COMMENTED** [review]
LGTM otherwise

File: src/fugw/mappings/sparse.py (hunk 2 above)
Comment (pbarbarant) [reply src/fugw/mappings/sparse.py:317]: I'm doing it ASAP.

File: examples/01_brain_alignment/plot_2_aligning_brain_sparse.py [hunk 3]
```diff
@@ -685,3 +685,53 @@ def plot_surface_map(
 )
 
 plt.show()
+
+# %%
+# We can also partially transport feature maps, by specifying
+# a parameter ``id_reg`` which balances the amount of regularization
+# applied between the tarnsport plan and the identity mapping. The
```
Comment (alexisthual) [inline examples/01_brain_alignment/plot_2_aligning_brain_sparse.py:692]:
```suggestion
# applied between the transport plan and the identity mapping. The
```

File: examples/01_brain_alignment/plot_2_aligning_brain_sparse.py [hunk 4]
```diff
@@ -685,3 +685,53 @@ def plot_surface_map(
...
+# regularized transport plan is then ``id_reg`` times the identity
+# plus ``(1 - id_reg)`` times the transport plan computed by FUGW.
+
+# Set of weights for the regularization parameter
+id_reg = 1 - np.linspace(0, 1, 3)
```
Comment (alexisthual) [inline examples/01_brain_alignment/plot_2_aligning_brain_sparse.py:697]: I'd be in favour of writing this explicitly like so:
```suggestion
id_reg = [.0, .5, 1.]
```

File: src/fugw/mappings/dense.py [hunk 5]
```diff
@@ -279,6 +279,10 @@ def transform(self, source_features, device="auto"):
         ----------
         source_features: ndarray(n_samples, n) or ndarray(n)
             Contrast map for source subject
+        id_reg: float, in the [0, 1] interval, defaults to 0
+            If source/target share the same geometry,
+            interpolate the transport plan with the identity
+            using the provided coefficient.
```
Comment (alexisthual) [inline src/fugw/mappings/dense.py:285]: Maybe add a description of what each value corresponds to:
```suggestion
            using the provided coefficient.
            A value of 1 (resp. 0) will rely solely on the identity
            (resp. the transport plan).
```

File: src/fugw/mappings/sparse.py [hunk 6]
```diff
@@ -314,6 +314,10 @@ def transform(self, source_features, device="auto"):
         ----------
         source_features: ndarray(n_samples, n) or ndarray(n)
             Contrast map for source subject
+        id_reg: float, in the [0, 1] interval, defaults to 0
+            If source/target share the same geometry,
+            interpolate the transport plan with the identity
+            using the provided coefficient.
```
Comment (alexisthual) [inline src/fugw/mappings/sparse.py:320]: Same here
```suggestion
            using the provided coefficient.
            A value of 1 (resp. 0) will rely solely on the identity
            (resp. the transport plan).
```

**Review by alexisthual — APPROVED** [review]
Some minor comments but it looks good to me :clap: 
Modifying one of the examples so that it showcases this feature might be easy to do yet very useful!
Thanks for this useful PR, I'm eager to see these results.

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Pull request merged**